src/cairo_runner.rs

```diff
@@ -144,6 +146,10 @@ impl PyCairoRunner {
             .initialize_segments(&mut self.pyvm.vm.borrow_mut(), None)
     }
 
+    pub fn segments(&self) -> PySegmentManager {
+        PySegmentManager::new(&self.pyvm, PyMemory::new(&self.pyvm))
```

Comment:
Couldn't this be an immutable reference? I'm not sure if the starknet code needs to read the segments at any time. I'm also not sure if it would want to even modify it

Reply:
🤔 both references are immutable